src/instructlab/configuration.py

```diff
@@ -29,23 +30,65 @@
 # Local
 from . import log
 
+
+class STORAGE_DIR_NAMES:
+    ILAB = "instructlab"
+    DATASETS = "datasets"
+    CHECKPOINTS = "checkpoints"
+    MODELS = "models"
+    TAXONOMY = "taxonomy"
+    INTERNAL = (
+        "internal"  # for storing all ilab-internal files the user doesn't need to see
+    )
+
+
+def get_home_dir(*subdir: str) -> str:
+    home = os.getenv("HOME")
+    if not home:
+        raise RuntimeError(
+            "The HOME directory does not exist. Please make sure it's set and run this again."
+        )
+    return os.path.join(home, *subdir)
+
+
+def get_home_env_path(env: str, suffix: str, *home_paths: str) -> str:
+    """Neatly formats the environment paths for the user home & storage directories"""
+    path = os.getenv(env) or get_home_dir(*home_paths)
+    return os.path.join(path, suffix)
+
+
+ILAB_CONFIG_HOME = get_home_env_path(
+    "XDG_CONFIG_HOME", STORAGE_DIR_NAMES.ILAB, ".config"
```

Comment:
`XDG_CONFIG_HOME` is for specifying where the python packages live?

Reply:
No, `XDG_CONFIG_HOME` specifies where we should store config files on the user's system. From the [FreeDesktop XDG Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):

> $XDG_CONFIG_HOME defines the base directory relative to which user-specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.

Reply:
The python packages will be installed in the `site-packages` of whichever python is used on the user's system, and `ilab` will be placed in its binaries folder.